builds/e2e/longhaul.yaml

```diff
@@ -344,3 +344,139 @@ jobs:
           test.deploymentFileName: 'long_haul_deployment_constrained.template.json'
           testResultCoordinator.storageAccountConnectionString: '$(EdgeLonghaulStorageAccountConnString)'
           clientModuleTransportType: 'mqtt'
+
+################################################################################
+  - job: Mariner1_amd64
+################################################################################
+    displayName: Mariner 1.0 AMD64
+    condition: and(eq(variables['run.EFLOW.amd64'], 'true'), eq(variables['useTRC'], 'true'))
+    pool:
+      name: $(pool.name)
+      demands:
+        - Agent.OS -equals Linux
+        - Agent.OSArchitecture -equals X64
+        - mariner -equals true
+        - mariner-version -equals 1.0
```

Comment:
Nit: We can prob can combine `mariner ` and `mariner-version` if we do `mariner -equals mariner1.0` or something.

Reply:
removed the mariner tag demand